subsys/net/ip/net_if.c

```diff
@@ -149,7 +149,9 @@ static inline void net_context_send_cb(struct net_context *context,
 
 static bool net_if_tx(struct net_if *iface, struct net_pkt *pkt)
 {
-	struct net_linkaddr *dst;
+	struct net_linkaddr *dst = NULL;
+	struct net_linkaddr ll_dst;
+	struct net_linkaddr_storage ll_dst_storage;
```

Comment:
I understand how this came up - from the desire to minimize needed changes, but looking closer, this 3-level pyramid - dst -> ll_dst -> ll_dst_storage - only adds more lines (and confusion) here. Please consider removing `dst` and using `ll_dst.addr != NULL` check instead of `(dst && dst->addr)`.

Reply:
Sure, this can be done like that too.